python_modules/libraries/dagster-github/dagster_github/resources.py

```diff
@@ -1,118 +1,159 @@
 import time
 from datetime import datetime
+from typing import Any, Dict, Optional
 
 import jwt
 import requests
-from dagster import Field, IntSource, StringSource, resource
+from dagster import ConfigurableResource, resource
+from dagster._config.structured_config import infer_schema_from_config_class
+from dagster._utils.cached_method import cached_method
+from pydantic import Field as PyField
 
 
 def to_seconds(dt):
     return (dt - datetime(1970, 1, 1)).total_seconds()
 
 
-class GithubResource:
-    def __init__(self, client, app_id, app_private_rsa_key, default_installation_id, hostname=None):
-        self.client = client
-        self.app_private_rsa_key = app_private_rsa_key
-        self.app_id = app_id
-        self.default_installation_id = default_installation_id
-        self.installation_tokens = {}
-        self.app_token = {}
-        self.hostname = hostname
+class GithubResource(ConfigurableResource):
+    github_app_id: int = PyField(
+        ...,
```

Comment:
should align on the use of ... as per https://github.com/dagster-io/dagster/pull/13232#discussion_r1157203715

Reply:
👍 responded in that thread, I'm ok standardizing with no ellipses